From 8c9a73c144ee85ea779f79d779da4908d08745ba Mon Sep 17 00:00:00 2001
From: tatou-tatou <tatou-tatou@users.noreply.github.com>
Date: Thu, 29 May 2014 18:16:41 +0200
Subject: [PATCH] Fix padding issue

---
 x.c | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/x.c b/x.c
index 0b40aa4..894d26f 100644
--- a/x.c
+++ b/x.c
@@ -219,6 +219,7 @@ static dimension_t calculate_dimensions(GSList *layouts)
 
         dim.h += (g_slist_length(layouts) - 1) * settings.separator_height;
         dim.h += g_slist_length(layouts) * settings.padding * 2;
+	dim.h += 2*settings.frame_width;
 
         int text_width = 0, total_width = 0;
         for (GSList *iter = layouts; iter; iter = iter->next) {
@@ -439,13 +440,10 @@ static dimension_t x_render_layout(cairo_t *c, colored_layout *cl, dimension_t d
 
         /* adding frame */
         bg_x += settings.frame_width;
-        if (first) {
-                bg_y += settings.frame_width;
-                bg_height -= settings.frame_width;
-        }
+        bg_y += settings.frame_width;
+        bg_height -= settings.frame_width;
         bg_width -= 2 * settings.frame_width;
-        if (last)
-                bg_height -= settings.frame_width;
+        bg_height += settings.frame_width;
 
         cairo_set_source_rgb(c, cl->bg.r, cl->bg.g, cl->bg.b);
         cairo_rectangle(c, bg_x, bg_y, bg_width, bg_height);
@@ -463,7 +461,7 @@ static dimension_t x_render_layout(cairo_t *c, colored_layout *cl, dimension_t d
         if (settings.separator_height > 0 && !last) {
                 cairo_set_source_rgb(c, sep_color.r, sep_color.g, sep_color.b);
 
-                cairo_rectangle(c, settings.frame_width, dim.y,
+                cairo_rectangle(c, settings.frame_width, dim.y + settings.frame_width,
                                 dim.w - 2 * settings.frame_width
                                 , settings.separator_height);
 
-- 
1.9.3

